charms/jupyter-controller/src/prometheus_alert_rules/host_resources.rule

```diff
@@ -0,0 +1,24 @@
+groups:
+- name: HostResources
+  rules:
+  - alert: FileDescriptorsExhausted
+    expr: process_open_fds / process_max_fds >= 0.98
```

Comment:
An arithmetic error (e.g. division by zero) results in `Empty query result.`, which means no alert would be fired. In this expression here it's low risk, but still I would change it to multiplication:
 ```suggestion
     expr: process_open_fds >= 0.98 * process_max_fds
```